### DISCUSSION TITLE
Maze Route cannot find path of net after custom macro placement

### DISCUSSION THREAD

**Discussion by Fahim36201** [The-OpenROAD-Project/OpenROAD-flow-scripts#911] [Q&A]
Changing macro placement causes: Maze Route cannot find path of net dmem/inter_dmem2 in worker of routeBox ( 79380 7560 ) ( 83160 11340 ).
Have tried changing top routing layer to M8
Also, `set_global_routing_layer_adjustment $env(MIN_ROUTING_LAYER)-$env(MAX_ROUTING_LAYER) 0.5`
And, `set_global_routing_layer_adjustment $env(MIN_ROUTING_LAYER)-$env(MAX_ROUTING_LAYER) 0.65`
Didn't help

To recreate the problem:

Design_nickname: riscv32i
Die_area: 0 0 120 120
Core_area: 5 5 110 110
Note: nothing changed in design/asap7/riscv32i/config.mk 

Macro placement target
Macro 1: name: dmem/dmem0, location: x0=15024 y0=10000 orientation: R0, place_status: FIRM
Macro 2: name: dmem/dmem1, location: x0=27505 y0=10000 orientation: R0, place_status: FIRM
Macro 3: name: dmem/dmem2, location: x0=82386 y0=10000 orientation: R0, place_status: FIRM
Macro 4: name: dmem/dmem3, location: x0=95867 y0=10000 orientation: R0, place_status: FIRM

**Comment by vijayank88** [comment]
@Fahim36201 
Your're saying changing macro placement causes issue in first line.
In recreate problem nothing changed in `config,mk`.
In flow tutorial video explained how to create reproducible test. Just follow that and attach the test case.

**Comment by Fahim36201** [comment]
@vijayank88 here are the log files of macro place and route.
[5_1_fastroute.log](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/11046967/5_1_fastroute.log)
[5_2_TritonRoute.log](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/11046968/5_2_TritonRoute.log)
[2_4_mplace.log](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/11046969/2_4_mplace.log)


here is the test case
https://drive.google.com/file/d/1SrlNCzfJV6jTdmgQ9pYUVUfmP8D8G16x/view?usp=sharing

**Reply by vijayank88** [reply]
@Fahim36201 Please share your `config.mk`

**Reply by Fahim36201** [reply]
here @vijayank88, changed them to text file uploading .mk is not permitted
![image](https://user-images.githubusercontent.com/126638432/227121744-05c0aec0-820b-4e8d-a255-19e9bcf184bc.png)


[config_platform.txt](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/11047653/config_platform.txt)

[config_design.txt](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/11047654/config_design.txt)

**Reply by vijayank88** [reply]
Have you tried without below variables?
```
#export REPAIR_TNS  = 100
export SETUP_SLACK_MARGIN = 100
export HOLD_SLACK_MARGIN = 100
export REMOVE_BUFFER_TREE = 1
export USE_GROUP_PATH = 1
```
What's your goal by using above variables?

**Reply by Fahim36201** [reply]
almost all were there by default, i only added RepairTns, SetupSlackMargin, HoldSlackMargin for various tests which worked with default macro palcement. GroupPath is for another test run which is also working and should not effext macros as it integrated in report_metrics.tcl
Do you suggest that i try without any of these variables and see if it works?

**Reply by vijayank88** [reply]
yes

**Reply by Fahim36201** [reply]
ok. let me give atry

**Reply by Fahim36201** [reply]
@vijayank88 still same 
[configdesign.txt](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/11047946/configdesign.txt)
[5_2_TritonRoute (1).log](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/files/11047947/5_2_TritonRoute.1.log)



testcase:
https://drive.google.com/file/d/1etlkLwL4TlIp4vNghUPLfpJG1PGkIZl_/view?usp=sharing

**Reply by vijayank88** [reply]
I just tried with my local installation. Not facing any issue. 
I think it is problem with your installation or setup.
ping me on slack channel with Direct Message

**Reply by vijayank88** [reply]
run `make clean_all` and run the flow again and check for `riscv32i`

**Reply by Fahim36201** [reply]
> I just tried with my local installation. Not facing any issue. I think it is problem with your installation or setup. ping me on slack channel with Direct Message

have you tried placing macros in locations other than the default one?

**Reply by vijayank88** [reply]
I was asking that `config.mk` and custom macro placement configuration file. You're not shared yet

**Reply by Fahim36201** [reply]
hi @vijayank88 knocked you personally in slack

**Comment by vijayank88** [comment]
For custom macro placement create `macro.cfg` to point which coordinates to place. Refer this [format](https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/flow/designs/sky130hd/microwatt/microwatt.macro_placement.cfg).
Add `macro.cfg` in `config.mk` with variable name `export MACRO_PLACEMENT=<path to>/macro.cfg`
Run the flow and check in OpenROAD gui, is macros placed as per configuration or not.

**Comment by vijayank88** [comment]
@Fahim36201 
Check this instructions on how to place macros manually to resolve your issue: https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/issues/916#issuecomment-1684029160

**Reply by Fahim36201** [reply]
ok. thanks.

**Comment by vijayank88** [comment]
It is discussed in #916 and closing this.
